launch_ros/launch_ros/utilities/to_parameters_list.py

```diff
@@ -28,25 +28,66 @@
 from ..parameters_type import EvaluatedParameters
 
 
+def __format_dict(dictionary, keys, formatted_dict):
+    """
+    Combine namespaces and their node names into one key and remove the "ros__parameters" key.
+
+    Return an empty dict if the "ros__parameters" key is not found.
+    """
+    for key, value in dictionary.items():
+        if key == 'ros__parameters':
+            node_name = '/'.join(keys)
+            formatted_dict[node_name] = value
+            return formatted_dict
+        elif type(value) is dict:
+            keys.append(key.lstrip('/'))
+            formatted_dict = __format_dict(value, keys, formatted_dict)
+            keys = []
+    return formatted_dict
+
+
 def to_parameters_list(
     context: LaunchContext,
+    node_name: str,
+    namespace: str,
     evaluated_parameters: EvaluatedParameters
 ) -> List[rclpy.parameter.Parameter]:
     """
     Transform evaluated parameters into a list of rclpy.parameter.Parameter objects.
 
     :param context: to carry out substitutions during normalization of parameter files.
         See `normalize_parameters()` documentation for further reference.
+    :param node_name: node name
+    :param namespace: node namespace
     :param parameters: parameters as either paths to parameter files or name/value pairs.
         See `evaluate_parameters()` documentation for further reference.
     :returns: a list of parameters
     """
     parameters = []  # type: List[rclpy.parameter.Parameter]
+    node_name = node_name.lstrip('/')
+    if not node_name:
+        print('Warning: no node name was provided. Parameter file entries that contain a '
+              'node name will not be applied')
```

Comment:
@rebecca-butler hmm, we should probably emit a warning if and only if we run into that situation. Also, consider using [`warnings`](https://docs.python.org/3.8/library/warnings.html) for warnings :)

Reply:
See 1f1ed6e